rpc/src/module/test.rs

```diff
@@ -111,6 +119,49 @@ impl IntegrationTestRpc for IntegrationTestRpcImpl {
         Ok(())
     }
 
+    fn generate_block(
+        &self,
+        block_assembler_script: Option<Script>,
+        block_assembler_message: Option<JsonBytes>,
+    ) -> Result<H256> {
+        let tx_pool = self.shared.tx_pool_controller();
+        let block_assembler_config = block_assembler_script.map(|script| BlockAssemblerConfig {
+            code_hash: script.code_hash,
+            hash_type: script.hash_type,
+            args: script.args,
+            message: block_assembler_message.unwrap_or_default(),
+        });
+        let block_template = tx_pool
+            .get_block_template_with_block_assembler_config(
+                None,
+                None,
+                None,
+                block_assembler_config,
+            )
+            .map_err(|err| RPCError::custom(RPCError::Invalid, err.to_string()))?
+            .map_err(|err| RPCError::custom(RPCError::CKBInternalError, err.to_string()))?;
+
+        let block: packed::Block = block_template.into();
+        let block_view = Arc::new(block.into_view());
+        let content = packed::CompactBlock::build_from_block(&block_view, &HashSet::new());
+        let message = packed::RelayMessage::new_builder().set(content).build();
+
+        // insert block to chain
+        self.chain
+            .process_block(Arc::clone(&block_view))
+            .map_err(|err| RPCError::custom(RPCError::CKBInternalError, err.to_string()))?;
+
+        // announce new block
```

Comment:
I think for integration test, it's better to add a option to control whether to broadcast the block, and add a new RPC to just broadcast the current tip. However, it is out of the scope of the current PR.